TrigScint/src/TrigScint/Firmware/hitproducer_hw.cxx

```diff
@@ -0,0 +1,167 @@
+#include <stdio.h>
+
+#include <iostream>
+
+#include "TrigScint/Firmware/hitproducer.h"
+#include "TrigScint/Firmware/objdef.h"
+
+void hitproducer_hw(ap_uint<14> FIFO[NHITS][5], Hit outHit[NHITS],
+                    ap_uint<8> Peds[NHITS]) {
+#pragma HLS ARRAY_PARTITION variable = FIFO complete
```

Comment:
These `pragma`s are likely an issue. I encountered a warning that they were undefined when compiling and so I wrapped the ones currently on `trunk` in a `#ifdef` to avoid them during compilation.

https://github.com/LDMX-Software/ldmx-sw/blob/ebbf27abb5b46398e8539c7ab02cfd9e77adcc69/TrigScint/src/TrigScint/Firmware/trackproducer_hw.cxx#L10-L22

Is this the proper way to do that? These `pragma` makes me think these lines of code are actually compiled by some other tool into some other product. Is that accurate? In which case, how can we make sure they are compiling as intended in this other environment?

Reply:
done